src/frontend/next/src/components/Posts/Timeline.tsx

```diff
@@ -0,0 +1,57 @@
+import { Container, createStyles, Grid } from '@material-ui/core';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import PostComponent from '../Post';
+import { Post } from '../../interfaces/index';
+import Spinner from '../Spinner';
+import LoadAutoScroll from './LoadAutoScroll';
+import useSiteMetaData from '../../hooks/use-site-metadata';
+
+type Props = {
+  pages: Post[][] | undefined;
+  nextPage: Function;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      padding: 0,
+      maxWidth: '785px',
+    },
+    activeCircle: {
+      borderRadius: '4rem',
+      transition: 'all linear 250ms',
+      color: theme.palette.primary,
+    },
+  })
+);
+
+const Timeline = ({ pages, nextPage }: Props) => {
+  const classes = useStyles();
+  const { telescopeUrl } = useSiteMetaData();
+
+  if (!(pages && pages.length)) {
+    return (
+      <Grid container spacing={0} direction="column" alignItems="center" justify="center">
+        <Spinner />
+      </Grid>
+    );
+  }
+
+  // Iterate over all the pages (an array of arrays) and then convert all post
+  // elements to <Post>
+  const postsTimeline = pages.map((page: Post[]): any =>
```

Comment:
`:any` isn't ideal.  Does it need typing (can it not infer it here)?

Not critical, but we could improve here or in follow-up.

Reply:
It seems like typescript doesn't like it without `:any`:
![image](https://user-images.githubusercontent.com/45835736/105436265-eed22d00-5c2c-11eb-9530-d6f06d01d975.png)
Not sure what to do.

Reply:
@rogercyyu : I would say keep any for now.

Reply:
I think the type here is
```ts
const postsTimeline = pages.map((page: Post[]): JSX.Element[] | JSX.Element =>
...
```

We should avoid using `any`. The whole point of using typescript is having type checking, but using `any` defeats that purpose.

Reply:
Problem is for `JSX.Element[]`, I get `'JSX' is not defined.eslintno-undef` unless we can use `// eslint-disable-next-line no-undef` to get rid of that.
![image](https://user-images.githubusercontent.com/45835736/105444021-0fa17f00-5c3b-11eb-80d2-94791ad663fd.png)
hmmm...

Reply:
@rogercyyu : Can you try sth like this Array<ReactElement>?

Reply:
ReactElement basically JSX.Element

Reply:
@tonyvugithub thanks, that fixed it.